packages/forma-36-react-components/src/components/ControlledInput/ControlledInput.tsx

```diff
@@ -34,72 +34,74 @@ const defaultProps: Partial<ControlledInputPropTypes> = {
   willBlurOnEsc: true,
 };
 
-export class ControlledInput extends Component<ControlledInputPropTypes> {
-  static defaultProps = defaultProps;
+export const ControlledInput = (props: ControlledInputPropTypes) => {
+  const {
+    className,
+    id,
+    testId,
+    required,
+    disabled,
+    onFocus,
+    onBlur,
+    name,
+    onChange,
+    checked,
+    value,
+    type,
+    labelText,
+    willBlurOnEsc,
+    ...otherProps
+  } = props;
 
-  handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
-    const ESC = 27;
+  const classNames = cn(styles['ControlledInput'], className, {
+    [styles['ControlledInput--disabled']]: disabled,
+  });
 
-    if (e.keyCode === ESC && this.props.willBlurOnEsc) {
-      e.currentTarget.blur();
-    }
-  };
+  const handleKeyDown = useCallback(
+    (e: KeyboardEvent<HTMLInputElement>) => {
+      e.persist();
```

Comment:
Is `e.persist()` a good idea here? react throws many warnings without it.
React peeps say they will get rid of [pooling](https://reactjs.org/docs/events.html) and `e.persist()` will no longer be needed in react `17.x`

Reply:
Yes it's fine here, and even though it doesn't do anything in React 17 it shouldn't complain, so we can keep it and stay compatible with both 16 and 17 👍